share/pulse/examples/Quicksort.Task.fst

```diff
@@ -51,39 +47,44 @@ fn rec t_quicksort
   if (lo < hi - 1)
   {
     let r = partition_wrapper a lo hi lb rb;
-    let pivot = p33 r;
-    with s1. assert (A.pts_to_range a lo r._1 s1);
-    with s2. assert (A.pts_to_range a r._1 r._2 s2);
-    with s3. assert (A.pts_to_range a r._2 hi s3);
+    let Mktuple3 p31 p32 pivot = r;
+    rewrite each r._1 as p31;
+    rewrite each r._2 as p32;
+    rewrite each r._3 as pivot;
+    // ^FIXME: would be nicer to rewrite r as (p31, p32, pivot) but projectors don't unfold
```

Comment:
Yes indeed.  Are you saying that `(x, y)._1` doesn't unify with `x`?  @nikswamy ?

Reply:
Not without unfolding, and we disable all unfolding during the unification checks performed by the matcher since we found them to go haywire previously. Probably something to revisit now.

I did add special support in Pulse to simplify `(x,y)._1 ~> x` for tuple2, I think I can extend this for all tuples and make this work, trying.

Reply:
I allowed some unfolding and now this is just `let p31, p32, pivot = partition_wrapper a lo hi lb rb;`